src/libopensc/pkcs15-openpgp.c

```diff
@@ -372,7 +372,7 @@ sc_pkcs15emu_openpgp_init(sc_pkcs15_card_t *p15card)
 	}
 
 	/* PKCS#15 DATA object from OpenPGP private DOs */
-	sc_pkcs15emu_openpgp_add_data(p15card);
+	r = sc_pkcs15emu_openpgp_add_data(p15card);
```

Comment:
@frankmorgner Yes, before, it was deliberately ignored. Because "Private DO" doesn't have many need (only requested by TrueCrypt use), I don't want the failure of initialization of "Private DO" make the whole process fail.

Reply:
Thanks for the details. I took out that change.

Reply:
@dpward Could you please add a comment to that line, so that we don't have to wonder later.

Reply:
@hongquan I updated the comment above this line, and a couple of others (after a glance at the [specification](https://gnupg.org/ftp/specs/OpenPGP-smart-card-application-3.3.pdf) to try to use the right wording).